packages/fxa-payments-server/src/components/CouponForm/index.tsx

```diff
@@ -15,23 +15,30 @@ import sentry from '../../lib/sentry';
 
 import * as Amplitude from '../../lib/amplitude';
 import { useCallbackOnce } from '../../lib/hooks';
+import { getMinimumAmount } from 'fxa-shared/subscriptions/stripe';
 
 /*
  * Check if the coupon promotion code provided by the user is valid.
  * If it is valid, return the discounted amount in cents.
  */
 const checkPromotionCode = async (planId: string, promotionCode: string) => {
   try {
-    const { discount } = await apiInvoicePreview({
+    const { discount, total, currency } = await apiInvoicePreview({
       priceId: planId,
       promotionCode,
     });
 
     if (!discount?.amount) {
       throw new Error('No discount for coupon');
     }
+
+    if (total < getMinimumAmount(currency) && total !== 0) {
+      throw new Error('Total below minimum amount allowed.');
+    }
+
     return discount.amount;
   } catch (err) {
+    console.log(err);
```

Comment:
Delete?

Reply:
One day I'll get a PR up without any console.logs...not today though.